src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -286,29 +285,20 @@ class StubGenerator: public StubCodeGenerator {
       // to frame manager / native entry.
       // Access all locals via frame pointer, because we know nothing about
       // the topmost frame's size.
-      __ ld_PPC(r_entryframe_fp, _abi_PPC(callers_sp), R1_SP_PPC);
-      assert_different_registers(r_entryframe_fp, R3_RET_PPC, r_arg_result_addr, r_arg_result_type, r_cr, r_lr);
-      __ ld_PPC(r_arg_result_addr,
-            _entry_frame_locals(result_address), r_entryframe_fp);
-      __ ld_PPC(r_arg_result_type,
-            _entry_frame_locals(result_type), r_entryframe_fp);
-      __ ld_PPC(r_cr, _abi_PPC(cr), r_entryframe_fp);
-      __ ld_PPC(r_lr, _abi_PPC(lr), r_entryframe_fp);
-
-      // pop frame and restore non-volatiles, LR and CR
-      __ mr_PPC(R1_SP_PPC, r_entryframe_fp);
-      __ mtcr_PPC(r_cr);
-      __ mtlr_PPC(r_lr);
-
-      // Store result depending on type. Everything that is not
-      // T_OBJECT, T_LONG, T_FLOAT, or T_DOUBLE is treated as T_INT.
-      __ cmpwi_PPC(CCR0, r_arg_result_type, T_OBJECT);
-      __ cmpwi_PPC(CCR1, r_arg_result_type, T_LONG);
-      __ cmpwi_PPC(CCR5, r_arg_result_type, T_FLOAT);
-      __ cmpwi_PPC(CCR6, r_arg_result_type, T_DOUBLE);
+
+      assert_different_registers(R10_RET1, r_arg_result_addr, r_arg_result_type);
+
+      __ ld(r_arg_result_addr, R8_FP, _entry_frame_locals(result_address));
+      __ ld(r_arg_result_type, R8_FP, _entry_frame_locals(result_type));
+
+
+      // pop frame and restore non-volatiles, SP and FP
+      __ mv(R2_SP, R8_FP);
+      __ ld(R1_RA, R8_FP, _abi(ra));
+      __ ld(R8_FP, R8_FP, _abi(fp));
```

Comment:
`pop_C_frame(true)`?

Reply:
It is an entry frame, not C. I don't want to mislead the reader. It is better to change in this way:
```
      __ mv(R2_SP, R8_FP);
      __ restore_abi_frame(R8_FP, 0);
```